config/config.go

```diff
@@ -119,21 +109,20 @@ type Metrics struct {
 }
 
 // OpenTelemetry collector configuration for tracing
-type OtelCollector struct {
+type TLSConfig struct {
 	CAName     string `yaml:"ca_name,omitempty"`
-	Protocol   string `yaml:"protocol,omitempty"` // http or https or grpc
 	SkipVerify bool   `yaml:"skip_verify,omitempty"`
-	TLSEnabled bool   `yaml:"tls_enabled,omitempty"`
 }
 
 // Tracing provides tracing configuration for the Kiali server.
 type Tracing struct {
-	CollectorType TracingCollectorType `yaml:"collector_type,omitempty"` // Possible values "otel" or "jaeger"
-	CollectorURL  string               `yaml:"collector_url,omitempty"`  // Endpoint for Kiali server traces
-	Enabled       bool                 `yaml:"enabled,omitempty"`
-	Otel          OtelCollector        `yaml:"otel,omitempty"`
+	CollectorURL string `yaml:"collector_url,omitempty"` // Endpoint for Kiali server traces
+	Enabled      bool   `yaml:"enabled,omitempty"`
+	Protocol     string `yaml:"protocol,omitempty"` // http or https or grpc
 	// Sampling rate for Kiali server traces. >= 1.0 always samples and <= 0 never samples.
-	SamplingRate float64 `yaml:"sampling_rate,omitempty"`
+	SamplingRate float64   `yaml:"sampling_rate,omitempty"`
+	TLSEnabled   bool      `yaml:"tls_enabled,omitempty"`
```

Comment:
To be consistent with other config options in Kiali, I think this should go inside `TLSConfig` as `TLSConfig.Enabled`. The pattern seems to be:

If it's a singular flag, it is named `<Flag>Enabled` e.g. `GzipEnabled` https://github.com/kiali/kiali/blob/c8210e66953dd09824b61f006403358fdead3f42/config/config.go#L150

If it's a struct with multiple options, `Enabled` goes inside the struct e.g. `Tracing.Enabled`: https://github.com/kiali/kiali/blob/c8210e66953dd09824b61f006403358fdead3f42/config/config.go#L133

A different way to do this is to omit the `Enabled` field altogether and instead have `Tracing.TLSConfig` be a pointer and if the pointer is non-nil then it is enabled implicitly. But Kiali doesn't follow this pattern for the rest of the config so I'd just stick with `TLSConfig.Enabled`.